### ISSUE TITLE
How to style using Styled Components

### ISSUE THREAD

**Issue by ezeikel** [fkhadra/react-toastify#342]
Are there any examples where `<ToastContainer />` can be styled using `styled-components`?

Currently doing this:

```
const ReactToastAdapter = ({ className, bodyClassName, progressClassName, ...props }) => {
  return  (
    <ToastContainer
      className={className}
      bodyClassName={bodyClassName}
      progressClassName={progressClassName}
      {...props}
    />
  )
};

const StyledToastContainer = styled(ReactToastAdapter).attrs({
  bodyClassName: 'body',
  progressClassName: 'progress',
})`
  /* .Toastify__toast-container */
  bottom: 0;
  left: 0;
  padding: 0;
  margin: 0;
  width: 100%;
  .Toastify__close-button {
    display: none;
  }
  .Toastify__toast {
    background-color: var(--color-black);
    margin: 0;
    cursor: auto;
  }
  .toast {
    background-color: var(--color-black);
  }
  .body {
    background-color: var(--color-black);
    color: var(--color-white);
    font-family: var(--default-font-family);
  }
`;
```

**Comment by ezeikel** [comment]
This seemed to do the trick:

```
const StyledToastContainer = styled(ToastContainer).attrs({
  className: 'toast-container',
  toastClassName: 'toast',
  bodyClassName: 'body',
  progressClassName: 'progress',
})`
  /* .toast-container */
  bottom: 0;
  left: 0;
  padding: 0;
  margin: 0;
  width: 100%;
  .toast {
    background-color: var(--color-black);
    margin: 0;
    cursor: auto;
  }
  button[aria-label="close"] {
    display: none;
  }
  .toast {
    background-color: var(--color-black);
  }
  .body {
    background-color: var(--color-black);
    color: var(--color-white);
    font-family: var(--default-font-family);
    margin: 0;
    display: grid;
    align-items: center;
  }
`;
```

**Comment by ezeikel** [comment]
Still getting this warning in the console though 🤔:

<img width="1440" alt="Screenshot 2019-04-19 at 19 55 28" src="https://user-images.githubusercontent.com/2964757/56439196-2fc14300-62dd-11e9-8069-7ca46128ed53.png">

**Comment by JrFarias** [comment]
The styled-component verify your children component that be wrapped to pass your classes.

https://github.com/styled-components/styled-components/blob/master/packages/styled-components/src/utils/classNameUsageCheckInjector.js#L41

However as the child is a div that does not have the classes passed by the styled-component this warning occurs.
https://github.com/fkhadra/react-toastify/blob/master/src/components/ToastContainer.js#L422

**Comment by fkhadra** [comment]
Hey,

Sorry for the late reply. Wanna make a PR to make the `ToastContainer` work with `styled-components` ?

**Comment by JrFarias** [comment]
Yeah, I'll fork this project and send a PR with the modification asap.

**Comment by sgoudie** [comment]
@JrFarias Hitting the same problem. Did you work on a fix for this?

**Comment by alextiley** [comment]
For those interested, you can work-around the warning mentioned in @ezeikel last comment by creating a wrapper component and applying the className to that instead:

```jsx
import React from 'react';
import styled from 'styled-components';
import { ToastContainer, ToastContainerProps } from 'react-toastify';

export const WrappedToastContainer = ({ className, ...rest }: ToastContainerProps & { className?: string }) => (
  <div className={className}>
    <ToastContainer {...rest} />
  </div>
);

export default styled(WrappedToastContainer).attrs({
  // custom props
})`
  .Toastify__toast-container {}
  .Toastify__toast {}
  .Toastify__toast--error {}
  .Toastify__toast--warning {}
  .Toastify__toast--success {}
  .Toastify__toast-body {}
  .Toastify__progress-bar {}
`;
```

**Comment by PichuPerez** [comment]
Is this a feature already?

**Comment by fkhadra** [comment]
https://fkhadra.github.io/react-toastify/how-to-style/#how-to-style-with-styled-components

**Comment by spookyvert** [comment]
@ezeikel @JrFarias @fkhadra Hey is this updated? I cannot override any existing CSS classes at all

**Comment by AriHrannar** [comment]
> https://fkhadra.github.io/react-toastify/how-to-style/#how-to-style-with-styled-components

It doesnt seem like this link works anymore? It just brings me to the demo page - I would expect that it should be a special page with code examples of how to style with styled-components?

**Comment by fkhadra** [comment]
@AriHrannar can you try from a private tab ? I suspect a cache issue

**Comment by AriHrannar** [comment]
> @AriHrannar can you try from a private tab ? I suspect a cache issue

Awesome, thanks that did it :)

**Comment by nemonemi** [comment]
The issue that I'm having is defining a colored theme overwrites the actual styled-components theme property.

```ts
import { ToastContainer as Toaster } from 'react-toastify';

export const ToastContainer = styled(Toaster).attrs(() => ({
  theme: 'colored',
}))`
  .Toastify__toast--warning {
    &,
    button {
      color: ${({ theme }) => theme.palette.text.primary}; // the theme is "colored" not the actual MUI theme that is being passed to the styled-components
    }
  }
...
```

Has anyone found a way around this?

**Comment by phoniks** [comment]
If anyone else is having issues with this and is working with both typescript and styled-components, I found that I wasn't getting my styles passed through because I hadn't told typescript to expect a className on my components.  Once I fixed that my custom styles started showing up.  This is the[stack over flow answer](https://stackoverflow.com/questions/54113367/extending-styles-with-styled-components-not-working) that made it clear to me what I was doing wrong - but @alextiley's example above also shows the proper way to handle this.  It might help to have a note to this effect in the documentation.
